### DISCUSSION TITLE
关于重定向 URI 丢失或不匹配问题的一些研究与解决方案

### DISCUSSION THREAD

**Discussion by lxdklp** [kookxiang/jellyfin-plugin-bangumi#109] [Show and tell]
我遇到了与 #100 一模一样的问题
![屏幕截图 2024-02-19 020058](https://github.com/kookxiang/jellyfin-plugin-bangumi/assets/69453767/8547c838-2304-49fa-b638-2e7390015bb7)
### 思路与解决方案
个人进行了猜测与实验，问题应该是出现在重定向 URI的回调地址上。我测试了如果jellyfin进行了反向代理就会造成uri不匹配，然后切换到服务器上用端口访问正常登录。
因此问题应该是出在这里
解决方案也很简单，登录时不使用反向代理即可
### 灵感来源
来自于之前我在经过反向代理的wac上登录Azure账户时也有同样的uri报错，来着微软的[问题解析与解决方案](https://learn.microsoft.com/zh-cn/troubleshoot/azure/active-directory/error-code-aadsts50011-redirect-uri-mismatch)
### 修改建议
建议加上禁止使用反向代理相关的提示

本文仅为个人研究，有错误欢迎和平指出

**Comment by chuyinluotian01** [comment]
感谢，删掉了代理果然登陆成功了

**Reply by G1a0** [reply]
还真是

**Comment by kookxiang** [comment]
https://jellyfin.org/docs/general/networking/nginx/
可以对比下官方文档看看 `X-Forwarded-***` 头有没有漏的，如果还不行可以更新下后台的 **网络 -  已知代理** 设置
